packages/records/lib/helpers/uniqueKey.ts

```diff
@@ -0,0 +1,48 @@
+import type { FormattedRecord } from '../types.js';
+
+export function getUniqueId(record: FormattedRecord): string {
+    return record.external_id;
+}
+
+export function verifyUniqueKeysAreUnique(records: FormattedRecord[]): { isUnique: boolean; nonUniqueKeys: string[] } {
+    const idMap: Record<string, boolean> = {};
+    let isUnique = true;
+    const nonUniqueKeys: string[] = [];
+
+    for (const record of records) {
+        const id = getUniqueId(record);
+
+        if (idMap[id]) {
+            isUnique = false;
+            if (!nonUniqueKeys.includes(id.toString())) {
+                nonUniqueKeys.push(id.toString());
+            }
+        } else {
+            idMap[id] = true;
+        }
+    }
+
+    return { isUnique, nonUniqueKeys };
+}
```

Comment:
Seems like `isUnique` is never used in the rest of the codebase. Also you could use Set for cleaner/faster dedup (not suggesting change)

```ts
export function verifyUniqueKeysAreUnique(records: FormattedRecord[]): { nonUniqueKeys: Set<string> } {
    const idMap = new Set<string>();
    const nonUniqueKeys = new Set<string>();

    for (const record of records) {
        const id = getUniqueId(record);
        if (idMap.has(id)) {
            nonUniqueKeys.add(id);
        } else {
            idMap.add(id);
        }
    }

    return { nonUniqueKeys };
}